enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,159 @@
+
+#### Story 2
+
+As an OpenShift administrator, I need visibility into which APIs that will be removed in the next EUS release are currently in use, so I can take any needed actions to ensure that I can upgrade to next EUS release.
+
+#### Story 3
+
+As an OpenShift administrator, I need visibilitiy into which workload is using an API that will be removed in the next release, so I can take any needed actions to ensure that I can upgrade to the next release. 
+
+### Implementation Details/Notes/Constraints
+
+#### Pre-release Lifecycle Background
+
+Pre-release APIs are now required to specify `prerelease-lifecycle-gen` tags that inform the apiserver to, among other things, maintain the `apiserver_requested_deprecated_apis` metric.
+
+```go
+// +k8s:prerelease-lifecycle-gen:introduced=1.1
+// +k8s:prerelease-lifecycle-gen:deprecated=1.8
+// +k8s:prerelease-lifecycle-gen:removed=1.18
+// +k8s:prerelease-lifecycle-gen:replacement=apps,v1,Deployment
+```
+
+The `apiserver_requested_deprecated_apis` metrics provides the metadata needed by this enhancement to "look into the future" and identify APIs, currently in use, that will be removed in the next release.
+
+In order to provide the metadata needed to look forward to the next EUS release, `prerelease-lifecycle-gen` tags updated between EUS releases will need to be back-ported to the latest EUS release.
+
+#### Alert Expression
+
+For the alert expression we filter the existing `apiserver_requested_deprecated_apis` series in combination with the `apiserver_request_total` series to determine if the API to be removed in the next release is actively in use.
+
+```
+group(apiserver_requested_deprecated_apis{removed_release="1.xx"}) by (group,version,resource)
+and
+(sum by(group,version,resource) (rate(apiserver_request_total[10m]))) > 0
+```
+
+Value of `removed_release`:
+
+| Openshift Release | removed_release | removed_release (EUS) |
+| ----------------- | --------------- | --------------------- |
+| 4.6 / 4.6 EUS     | 1.20            | 1.23                  |
+| 4.7               | 1.21            |                       |
+| 4.8               | 1.22            |                       |
+| 4.9               | 1.23            |                       |
+| 4.10 / 4.10 EUS   | 1.24            | TBD                   |
+
+
+#### Determinine Workload
+
+Documentation will be provided on how to use the audit logs to identify the workload using the API that will be removed.
+
+
+### Risks and Mitigations
+
+#### Storage Version Migration
+
+During an EUS upgrade, where multiple "normal" releases are applied back-to-back, there is a risk that there are persisted instances of resources that can longer be decoded from storage due to the removal of the storage encoder for a removed version. We should either provide customers with documentation on upgrading storage versions or provide a storage version migration tool to perform the migrations automatically.
```

Comment:
IMO kube-apiserver operator should make sure that this migration is done, behind the scenes.